omero_figure/views.py

```diff
@@ -228,17 +228,13 @@ def save_web_figure(request, conn=None, **kwargs):
 
         # Try to set Group context to the same as first image
         curr_gid = conn.SERVICE_OPTS.getOmeroGroup()
-        try:
-            conn.SERVICE_OPTS.setOmeroGroup('-1')
-            i = conn.getObject("Image", first_img_id)
-            if i is not None:
-                gid = i.getDetails().getGroup().getId().getValue()
-                conn.SERVICE_OPTS.setOmeroGroup(gid)
-            else:
-                # Don't leave as -1
-                conn.SERVICE_OPTS.setOmeroGroup(curr_gid)
-        except:
-            # revert back
+        conn.SERVICE_OPTS.setOmeroGroup('-1')
+        i = conn.getObject("Image", first_img_id)
+        if i is not None:
+            gid = i.getDetails().getGroup().getId()
```

Comment:
shouldn't it be ``getId().getValue()``?

Reply:
There's a couple of ways to do this. getGroup() returns a Wrapper, so that getId() doesn't need unwrapping:
```
In [3]: i = conn.getObject('Image', 73995)
In [4]: i.getDetails().getGroup().getId()
Out[4]: 5L
In [5]: i.getDetails().group.id.val
Out[6]: 5L
```

Reply:
Thanks